### PR TITLE
Golovinsky images

### PR DESCRIPTION
**Pull request by golxzn**

### CODE REVIEW

**Comment by seigtm** [comment]
Fuck me it's the best shit I've seen in ma life dude

File: TODO [hunk 1]
```diff
@@ -9,7 +9,20 @@
...
   [ ] Integrate color class with glm;
   [ ] Integrate angle class with glm;
+
+[ ] Platforms
+  [ ] MacOS
+  [ ] iOS
+  [ ] Android
+
+[ ] Tests
+  [ ] Make individual target Benchmarks and move benchs from tests
```
Comment (seigtm) [inline TODO]: NO NEWLINES == NO BITCHES

File: code/src/golxzn/core/types/color.cpp [hunk 2]
```diff
@@ -13,6 +13,12 @@ const u8 *const color::raw() const noexcept {
 	return reinterpret_cast<const u8 *const >(this);
 }
 
+u8 *color::begin() noexcept { return raw(); }
+u8 *color::end() noexcept { return raw() + 4; }
```
Comment (seigtm) [inline code/src/golxzn/core/types/color.cpp]: looks magical and creepy but ok

File: tests/assets/images/test.jpg
Comment (seigtm) [inline tests/assets/images/test.jpg:1]: S3XY

File: tests/sources/TestImage.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+// TEST_CASE("Image", "[golxzn][core][benchmarks]") {
+// }
+
+// int main(int argc, char* argv[]) {
+// 	Catch::Session session;
+// 	session.configData().benchmarkSamples = 10; // set number of iterations
+// 	int returnCode = session.run(argc, argv);
+// 	return returnCode;
+// }
```
Comment (seigtm) [inline tests/sources/TestImage.cpp]: NO NEWLINES == BO BITCHES (2)

File: tests/sources/TestResourceManager.cpp [hunk 4]
```diff
@@ -10,4 +14,62 @@ TEST_CASE("Resource Manager", "[golxzn][core][tests]") {
...
+// 	BENCHMARK("flip v: true,  h: true ")  { return for_bench.flip(true, true);  };
+// 	BENCHMARK("save_image: user://images/bench/test.tga") { return for_bench.save_image("user://images/bench/test.tga"); };
+// }
+
+// int main(int argc, char* argv[]) {
+// 	Catch::Session session;
+// 	session.configData().benchmarkSamples = 10; // set number of iterations
+// 	int returnCode = session.run(argc, argv);
+// 	return returnCode;
+// }
```
Comment (seigtm) [inline tests/sources/TestResourceManager.cpp]: NO NEWLINES == NO BITCHES (3)

File: TODO (hunk 1 above)
Comment (golxzn) [reply TODO]: FIXED

File: code/src/golxzn/core/types/color.cpp (hunk 2 above)
Comment (golxzn) [reply code/src/golxzn/core/types/color.cpp]: Now it's ok

**Review by seigtm — APPROVED** [review]
S3XY

**Pull request merged**